src/java/org/apache/cassandra/index/sai/disk/vector/EmptyScoreRowIdIterator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.vector;
+
+import org.apache.cassandra.utils.CloseableIterator;
+
+public class EmptyScoreRowIdIterator implements CloseableIterator<ScoredRowId>
```

Comment:
Let's add this to CloseableIterator instead?

```
public interface CloseableIterator<T> extends Iterator<T>, AutoCloseable
{
    public void close();

    static final CloseableIterator<Object> EMPTY = new CloseableIterator<>()
    {
        public void close() {}
        public boolean hasNext() { return false; }
        public Object next() { throw new NoSuchElementException(); }
    };

    @SuppressWarnings("unchecked")
    static <T> CloseableIterator<T> emptyIterator()
    {
        return (CloseableIterator<T>) EMPTY;
    }
}
```

The only thing I don't love is that EMPTY is public b/c of the rules for interfaces but I think I can live with that

Reply:
I'm good with that.

Reply:
Done (with the slight modification that I just called `CloseableIterator.wrap(Collections.emptyIterator())` for simplicity.